samplers/jaegerremote/sampler.go

```diff
@@ -67,24 +61,13 @@ func (s *probabilisticSampler) SamplingRate() float64 {
 }
 
 func (s *probabilisticSampler) ShouldSample(p trace.SamplingParameters) trace.SamplingResult {
-	psc := oteltrace.SpanContextFromContext(p.ParentContext)
-	traceID := binary.BigEndian.Uint64(p.TraceID[0:8])
-	if s.samplingBoundary >= traceID&maxRandomNumber {
-		return trace.SamplingResult{
-			Decision:   trace.RecordAndSample,
-			Tracestate: psc.TraceState(),
-		}
-	}
-	return trace.SamplingResult{
-		Decision:   trace.Drop,
-		Tracestate: psc.TraceState(),
-	}
+	return s.sampler.ShouldSample(p)
 }
 
 // Equal compares with another sampler.
 func (s *probabilisticSampler) Equal(other trace.Sampler) bool {
 	if o, ok := other.(*probabilisticSampler); ok {
-		return s.samplingBoundary == o.samplingBoundary
+		return s.samplingRate == o.samplingRate
```

Comment:
float comparison should be done with an epsilon

Reply:
ah yes, thanks for the reminder 🤦🏻